hydromt/data_adapter/dataframe.py

```diff
@@ -176,54 +178,88 @@ def to_file(
 
 
         """
-        kwargs.pop("bbox", None)
-        obj = self.get_data(time_tuple=time_tuple, variables=variables, logger=logger)
-
-        read_kwargs = dict()
-        if driver is None or driver == "csv":
-            # always write as CSV
-            driver = "csv"
-            fn_out = join(data_root, f"{data_name}.csv")
-            obj.to_csv(fn_out, **kwargs)
-            read_kwargs["index_col"] = 0
-        elif driver == "parquet":
-            fn_out = join(data_root, f"{data_name}.parquet")
-            obj.to_parquet(fn_out, **kwargs)
-        elif driver == "excel":
-            fn_out = join(data_root, f"{data_name}.xlsx")
-            obj.to_excel(fn_out, **kwargs)
-        else:
-            raise ValueError(f"DataFrame: Driver {driver} is unknown.")
+        try:
+            kwargs.pop("bbox", None)
+            obj = self.get_data(
```

Comment:
Could you here decide to pass the ``handle_nodata`` argument? Then get_data will take care or returning either error that you want or data then you write or None and then you don't write? Instead of again doing try/except?